### ISSUE TITLE
Graphile Pro: overrideOptionsForRequest defaultPaginationCap is not taken into account

### ISSUE THREAD

**Issue by crubier** [graphile/issues#7]
### Summary

`defaultPaginationCap` in `overrideOptionsForRequest` is not taken into account.

### Steps to reproduce

I use this config of Graphile Pro:

```
postgraphile( dbUrl, schemaName, {
  defaultPaginationCap: -1,
  graphqlDepthLimit: -1,
  graphqlCostLimit: -1,
  exposeGraphQLCost: false,
  overrideOptionsForRequest(req) {
    return {
      defaultPaginationCap: 2,
      graphqlDepthLimit: 2, 
      graphqlCostLimit: 10000,
      exposeGraphQLCost: false
    };
  }
});
```
 
### Expected results

- I run a query with a cost > 10000  => I get an error saying it is above the allow limit ✅ 
- I run a query with a depth > 2 => I get an error saying it is above the allow limit ✅ 
- I run a query with no pagination => I get an error saying it is above the allow limit ✅ 
- I run a query with first > 2 => I get an error saying it is above the allow limit ✅ 

### Actual results

- I run a query with a cost > 10000  => I get an error saying it is above the allow limit ✅ 
- I run a query with a depth > 2 => I get an error saying it is above the allow limit ✅ 
- I run a query with no pagination => Graphile let it pass while it shouldn't 🔴 
- I run a query with first > 2 => Graphile let it pass while it shouldn't 🔴 

### Additional context

More generally, the `defaultPaginationCap` returned by `overrideOptionsForRequest` is never taken into account. But the `defaultPaginationCap` from the config is well-enforced correctly. 

We use quite a lot of plugins, but I don't see anything that would interfere with that. Especially knowing that `defaultPaginationCap` from config is well-enforced.

Using `@graphile/pro@1.0.1` and `postgraphile@4.11.0` and `postgraphile@4.12.0-alpha.0`

### Possible Solution

No idea since the plugin is not open source, and the minified code is not so easy to debug 😬

**Comment by crubier** [comment]
@benjie :-)

**Comment by benjie** [comment]
Reproduced in the tests :+1:

**Comment by benjie** [comment]
Released fix in `@graphile/pro@1.0.2`

**Comment by crubier** [comment]
Perfect thanks so much for the reactivity! 👍
